apps/events-helsinki/src/domain/search/eventSearch/advancedSearch/TargetAgeGroupSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import { useSearchTranslation } from '@events-helsinki/components';
+import type { SingleSelectProps } from 'hds-react';
+import { IconGroup, Select } from 'hds-react';
+import { TARGET_GROUP_AGE_GROUPS_IN_ORDER } from '../../../../constants';
+
+/**
+ * Get select component options for target group age selection.
+ * @param addEmptyOption should an empty selection be included. Defaults to `true`.
+ * @returns a list of select options
+ */
+function useTargetAgeGroupSelectorOptions(addEmptyOption = true) {
+  const { t: tSearch } = useSearchTranslation();
+
+  const targetGroups = TARGET_GROUP_AGE_GROUPS_IN_ORDER.map((ageGroup) => ({
+    label: tSearch(`search.targetAgeGroup.${ageGroup}`),
+    value: ageGroup,
+  }));
+
+  if (addEmptyOption) {
+    return [{ label: '', value: '' }, ...targetGroups];
+  }
+  return targetGroups;
+}
+
+export type TargetAgeGroupOptionType = ReturnType<
+  typeof useTargetAgeGroupSelectorOptions
+>[number];
```

Comment:
Using something like this type here could bring more clarity/transparency to what this type is. And a bit more strictness as the current type does not specify that the value needs to be AgeGroup or empty string, but the return type is (probably, not 100% sure) resolved as `{label: string; value: string}`:
```typescript
type TargetAgeGroupOptionType = { label: string; value: AgeGroup | '' };
```
Not a big deal, but just was a bit curious to see what the type was and thought about to what specificity can Typescript see the return type's values (i.e. the use of map function is the one that makes the "as const" mapping lose information for the target group values AFAIK).

Reply:
It seems that other places don't accept the stricter definition of `value` field here.
So maybe just simplifying to:
```typescript
export type TargetAgeGroupOptionType = { label: string; value: string };
```

Reply:
I'd go with the simplest solution here for clarity, i.e. the OptionType and I'd give the function that should return it it's return type as OptionType[]. That's very simple, enforces that the function does return such a type, and clear for the user to see what the type is. But, I'm not going to force this issue here, I am giving my own recommendation, and you can do as you see is best here.